va/validation-authority.go

```diff
@@ -637,33 +641,85 @@ func (va *ValidationAuthorityImpl) checkCAARecords(ctx context.Context, identifi
 	if err != nil {
 		return
 	}
+
 	if caaSet == nil {
 		// No CAA records found, can issue
+		va.stats.Inc("VA.CAA.None", 1, 1.0)
 		present = false
 		valid = true
 		return
-	} else if caaSet.criticalUnknown() {
-		present = true
+	}
+
+	present = true
+
+	// Record stats on directives not currently processed.
+	if len(caaSet.Iodef) > 0 {
+		va.stats.Inc("VA.CAA.WithIodef", 1, 1.0)
+	}
+
+	if caaSet.criticalUnknown() {
+		// Contains unknown critical directives.
+		va.stats.Inc("VA.CAA.UnknownCritical", 1, 1.0)
 		valid = false
 		return
-	} else if len(caaSet.Issue) > 0 || len(caaSet.Issuewild) > 0 {
-		present = true
-		var checkSet []*dns.CAA
-		if strings.SplitN(hostname, ".", 2)[0] == "*" {
-			checkSet = caaSet.Issuewild
-		} else {
-			checkSet = caaSet.Issue
-		}
-		for _, caa := range checkSet {
-			if caa.Value == va.IssuerDomain {
-				valid = true
-				return
-			}
+	}
+
+	if len(caaSet.Unknown) > 0 {
+		va.stats.Inc("VA.CAA.WithUnknownNoncritical", 1, 1.0)
+	}
+
+	checkSet := caaSet.Issue
+	if strings.SplitN(hostname, ".", 2)[0] == "*" {
```

Comment:
Since `policy.WillingToIssue` won't let us get this far (wildcards are not allowed) we should just rip out the `checkSet` and issueWild check and always use `caaSet.Issue`.